### PR TITLE
`initMetrics`

### PR DESCRIPTION
**Pull request by jcp19**

### CODE REVIEW

**Review by Dspil — APPROVED** [review]

File: router/dataplane_spec.gobra [hunk 1]
```diff
@@ -75,18 +82,40 @@ pred AccBfdSession(bfdSessions map[uint16]bfdSession) {
 
 pred AccForwardingMetrics(metrics map[uint16]forwardingMetrics) {
 	acc(metrics, 1/2) &&
-	forall fm forwardingMetrics :: fm in range(metrics) ==> acc(forwardingMetricsMem(fm), _)
+	forall id uint16 :: id in domain(metrics) ==> acc(forwardingMetricsMem(metrics[id], id), _)
```
Comment (Dspil) [inline router/dataplane_spec.gobra:85]: can we add a trigger here or will it be invalid?

Comment (jcp19) [reply router/dataplane_spec.gobra]: good point, done

File: router/dataplane.go [hunk 2]
```diff
@@ -728,19 +733,75 @@ func (d *DataPlane) Run(ctx context.Context) error {
...
+	// @ invariant forall i uint16 :: { d.forwardingMetrics[i] } i in domain(d.forwardingMetrics) ==>
+	// @ 	acc(forwardingMetricsMem(d.forwardingMetrics[i], i), _)
+	// @ invariant acc(&d.Metrics, def.ReadL15)
+	// @ invariant acc(d.Metrics.Mem(), _)
+	// @ decreases len(d.external) - len(visitedSet)
+	for id := range d.external /*@ with visitedSet @*/ {
+		// @ ghost if d.internalNextHops != nil {
+		// @	unfold acc(AccAddr(d.internalNextHops), def.ReadL20)
+		// @ }
+		// @ assert d.internalNextHops[id] === d.internalNextHops[id]
```
Comment (jcp19) [inline router/dataplane.go:787]:
```suggestion
```

File: router/dataplane_spec.gobra [hunk 3]
```diff
@@ -18,12 +18,19 @@ package router
 
 import (
 	"net"
+	"github.com/scionproto/scion/pkg/addr"
+	// def "github.com/scionproto/scion/verification/utils/definitions"
```
Comment (jcp19) [inline router/dataplane_spec.gobra:22]:
```suggestion
```

**Pull request merged**